### ISSUE TITLE
`Content-Type:application/octet-stream` being set for subsequent middleware

### ISSUE THREAD

**Issue by noinkling** [shellscape/koa-webpack#23]
Already mentioned in #15 (and leecade/koa-webpack-middleware#23), but I thought I'd open a new issue.

`Content-Type:application/octet-stream` is being set for any middleware hit after this one. Besides being generally incorrect if you're trying to respond with html/text/json, it causes the browser (Chrome at least) to prompt for a download. Working around it means manually setting `ctx.type` to something more appropriate, or else making sure the webpack middleware only gets called on a route where it won't be a problem.

This behaviour is a symptom of `context.body` being unconditionally set to a stream (the passthrough) in the hot middleware part [here](https://github.com/shellscape/koa-webpack/blob/eba327e1d6aaf1e3f128e33bcbbeb28077a1a3b0/index.js#L52). As soon as `context.body` gets assigned, Koa sets the appropriate content type, which then remains even after later reassigning `body` to a string or whatever.

Side note: `writeHead` should be setting `context.status`, not `context.state`, it should look like this:

```js
writeHead: (status, headers) => {
  context.status = status;
  context.set(headers);
}
```
But the method that has been used to hook-in seems a bit hacky and brittle in general to me. I'll see if I can find the time to put together a PR, just wanted to raise this issue while it's in my head.

Thanks for creating this repo by the way, the other one isn't maintained very well 😛

**Comment by shellscape** [comment]
thanks for taking the time to triage this. I'll be sure to tag you in the PR if I get to it before you.

**Comment by noinkling** [comment]
I opened a PR with a minimal fix, but obviously things remain quite brittle. While I was trying to think of a way to solve this I came up with a few other ideas:

1. Duplicate [the path check from the original middleware](https://github.com/glenjamin/webpack-hot-middleware/blob/bd572e6e68939018542888b7a737157ed3c46c18/middleware.js#L26), only mess with the response and invoke the original middleware if it passes. Not sure that's much of an improvement over the PR though...

2. Ideally we could just pass in `context.res`. There's some evidence this works [here](https://github.com/glenjamin/webpack-hot-middleware/issues/20#issuecomment-159249958) and it [was included in the readme previously](https://github.com/glenjamin/webpack-hot-middleware/blob/v2.10.0/README.md#koa). But if it works then I'm not sure why @leecade would decide to do it the current way in the first place. In other words it probably needs plenty of tests.

3. Don't delegate at all, just rewrite the middleware for Koa. Obviously not desirable for various reasons.

4. Get the original middleware to include a Koa-compatible version, or maybe just add some helpful hooks. Problem with this is that the maintainer isn't very amenable to the idea: glenjamin/webpack-hot-middleware#123
